[package] yasm/1.3.0: CMake 4 generation fails

### Description

When a system has CMake 4 installed CMake generation fails with an error
`Compatibility with CMake < 3.5 has been removed from CMake`

### Package and Environment Details

* Package Name/Version: **yasm/1.3.0**
* Conan version: **conan 2.17.0**
* CMake version: **CMake 4.0.1**


### Conan profile

[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=msvc
compiler.cppstd=23
compiler.cstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows

### Steps to reproduce

conan install --build=missing -r conancenter .

### Logs

```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=msvc
compiler.cppstd=23
compiler.cstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=RelWithDebInfo
compiler=msvc
compiler.cppstd=23
compiler.cstd=17
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows


======== Computing dependency graph ========
yasm/1.3.0: Not found in local cache, looking in remotes...
yasm/1.3.0: Checking remote: conancenter
Connecting to remote 'conancenter' anonymously
yasm/1.3.0: Downloaded recipe revision fb800a15413dca19bfaef9e4b5d50694
Graph root
    conanfile.py: C:\Projects\conan-test\conanfile.py
Requirements
    yasm/1.3.0#fb800a15413dca19bfaef9e4b5d50694 - Downloaded (conancenter)

======== Computing necessary packages ========
Requirements
    yasm/1.3.0#fb800a15413dca19bfaef9e4b5d50694:e5c3e3fe37192d11f4aca55d0b81ad141342359d - Build

======== Installing packages ========
yasm/1.3.0: Sources downloaded from 'conancenter'
yasm/1.3.0: Calling source() in C:\Users\---\.conan2\p\yasmb78308884dd42\s\src
yasm/1.3.0: Uncompressing yasm-1.3.0.tar.gz to C:\Users\---\.conan2\p\yasmb78308884dd42\s\src

-------- Installing package yasm/1.3.0 (1 of 1) --------
yasm/1.3.0: Building from source
yasm/1.3.0: Package yasm/1.3.0:e5c3e3fe37192d11f4aca55d0b81ad141342359d
yasm/1.3.0: settings: os=Windows arch=x86_64 build_type=RelWithDebInfo
yasm/1.3.0: Copying sources to build folder
yasm/1.3.0: Building your package in C:\Users\---\.conan2\p\b\yasm3e5cb61d986d1\b
yasm/1.3.0: Calling generate()
yasm/1.3.0: Generators folder: C:\Users\---\.conan2\p\b\yasm3e5cb61d986d1\b\build\generators
yasm/1.3.0: CMakeToolchain generated: conan_toolchain.cmake
yasm/1.3.0: CMakeToolchain generated: C:\Users\---\.conan2\p\b\yasm3e5cb61d986d1\b\build\generators\CMakePresets.json
yasm/1.3.0: CMakeToolchain generated: C:\Users\---\.conan2\p\b\yasm3e5cb61d986d1\b\src\CMakeUserPresets.json
yasm/1.3.0: Generating aggregated env files
yasm/1.3.0: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
yasm/1.3.0: Calling build()
yasm/1.3.0: Apply patch (conan): CMake 3.0+ compatibility
yasm/1.3.0: Running CMake.configure()
yasm/1.3.0: RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/---/.conan2/p/b/yasm3e5cb61d986d1/p" -DYASM_BUILD_TESTS="OFF" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/Users/---/.conan2/p/b/yasm3e5cb61d986d1/b/src"
CMake Error at CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


-- Configuring incomplete, errors occurred!

yasm/1.3.0: ERROR:
Package 'e5c3e3fe37192d11f4aca55d0b81ad141342359d' build failed
yasm/1.3.0: WARN: Build folder C:\Users\---\.conan2\p\b\yasm3e5cb61d986d1\b\build
ERROR: yasm/1.3.0: Error in build() method, line 85
        cmake.configure()
        ConanException: Error 1 while executing
```

--- a/recipes/yasm/all/conanfile.py
+++ b/recipes/yasm/all/conanfile.py
@@ -41,6 +41,8 @@ class YASMConan(ConanFile):
         del self.info.settings.compiler
 
     def build_requirements(self):
+        self.tool_requires("cmake/[>=3.0 <4]")
+
         if self._settings_build.os == "Windows" and not is_msvc(self):
             self.win_bash = True
             if not self.conf.get("tools.microsoft.bash:path", check_type=str):

--- a/recipes/yasm/all/conanfile.py
+++ b/recipes/yasm/all/conanfile.py
@@ -7,7 +7,7 @@ from conan.tools.layout import basic_layout
 from conan.tools.microsoft import is_msvc
 import os
 
-required_conan_version = ">=1.54.0"
+required_conan_version = ">=2.4"
 
 
 class YASMConan(ConanFile):
@@ -19,17 +19,10 @@ class YASMConan(ConanFile):
     topics = ("yasm", "installer", "assembler")
     license = "BSD-2-Clause"
     settings = "os", "arch", "compiler", "build_type"
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    languages = ["C"]
 
     def export_sources(self):
         export_conandata_patches(self)
-
-    def configure(self):
-        self.settings.rm_safe("compiler.cppstd")
-        self.settings.rm_safe("compiler.libcxx")
 
     def layout(self):
         if is_msvc(self):
@@ -41,9 +34,8 @@ class YASMConan(ConanFile):
         del self.info.settings.compiler
 
     def build_requirements(self):
-        self.tool_requires("cmake/[>=3.0 <4]")
-
-        if self._settings_build.os == "Windows" and not is_msvc(self):
+        self.tool_requires("cmake/[>=4]")
+        if self.settings_build.os == "Windows" and not is_msvc(self):
             self.win_bash = True
             if not self.conf.get("tools.microsoft.bash:path", check_type=str):
                 self.tool_requires("msys2/cci.latest")
@@ -108,7 +100,3 @@ class YASMConan(ConanFile):
     def package_info(self):
         self.cpp_info.includedirs = []
         self.cpp_info.libdirs = []
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info(f"Appending PATH environment variable: {bin_path}")
-        self.env_info.PATH.append(bin_path)

--- a/recipes/yasm/all/conanfile.py
+++ b/recipes/yasm/all/conanfile.py
@@ -34,7 +34,6 @@ class YASMConan(ConanFile):
         del self.info.settings.compiler
 
     def build_requirements(self):
-        self.tool_requires("cmake/[>=4]")
         if self.settings_build.os == "Windows" and not is_msvc(self):
             self.win_bash = True
             if not self.conf.get("tools.microsoft.bash:path", check_type=str):

--- a/recipes/yasm/all/patches/0001-cmake-updates.patch
+++ b/recipes/yasm/all/patches/0001-cmake-updates.patch
@@ -1,7 +1,7 @@
 --- a/CMakeLists.txt
 +++ b/CMakeLists.txt
 @@ -1,5 +1,5 @@
-+CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
++CMAKE_MINIMUM_REQUIRED(VERSION 3.5)
  PROJECT(yasm)
 -CMAKE_MINIMUM_REQUIRED(VERSION 2.4)
  if (COMMAND cmake_policy)

